### ISSUE TITLE
"ERROR  >>  errexit on line 386" when running on virus data 

### ISSUE THREAD

**Issue by boweny920** [padlocbio/padloc#16]
Hi , the below error popped up for some of my viral fna data. The tool seems to be working for some fna data but not for others. Is this caused by the nature of the data or the tool itself?  Thanks!! 

Backtrace:
    █
 1. ├─gff %>% filter(type == "CDS") %>% separate_attributes()
 2. ├─global::separate_attributes(.)
 3. │ └─`%>%`(...)
 4. ├─tidyr::spread(., key = key, value = value, fill = NA)
 5. └─tidyr:::spread.data.frame(., key = key, value = value, fill = NA)
Execution halted
(13:15:04) ERROR  >>  errexit on line 386

**Comment by JacksonLab** [comment]
Hello,
Do you have an example file that fails you wouldn't mind sharing? Cheers,
Simon

**Comment by boweny920** [comment]
Of course. I have attached an example file below for your reference, and thanks much for the prompt reply!
[example_file.fna.zip](https://github.com/padlocbio/padloc/files/7915176/example_file.fna.zip)

Bowen

**Comment by JacksonLab** [comment]
Hi Bowen, 

Apologies for the delay, slipped off the radar... I ran a test and the traceback earlier goes:

[03:13:22 PM] DEBUG >> Reading hmm_meta.txt
[03:13:22 PM] DEBUG >> Reading sys_meta.txt
[03:13:22 PM] DEBUG >> Reading c7b4b974-b55a-41ff-8996-4cc499720395.domtblout
[03:13:26 PM] DEBUG >> Reading c7b4b974-b55a-41ff-8996-4cc499720395_prodigal.gff
Error: Each row of output must be identified by a unique combination of keys.
Keys are shared for 47138 rows:
* 2412, 32104
* 2425, 32117
* 2438, 32130...

I checked the input fasta example you gave and there are multiple sequences (likely duplicates) with the same fasta IDs. For example ">UGV-GENOME-0115401" is in there three times. I'll add another input parsing step that enforces unique contig/sequence names and warns when this condition is not met.

Cheers,
Simon
